Makefile

```diff
@@ -20,11 +19,15 @@ $(UTILS)/cpuinfo.o : $(UTILS)/cpuinfo.cpp
 $(TESTDIR)/%.o : $(TESTDIR)/%.cpp $(SRCS)
 		$(CXX) $(CXXFLAGS) $(MARCHFLAG) $(GTESTCFLAGS) -c $< -o $@
 
-test: $(TESTDIR)/main.cpp $(TESTOBJS) $(UTILS)/cpuinfo.o $(SRCS)
-		$(CXX) tests/main.cpp $(TESTOBJS) $(UTILS)/cpuinfo.o $(MARCHFLAG) $(CXXFLAGS) $(GTESTLDFLAGS) -o testexe
+test: $(TESTOBJS) $(UTILS)/cpuinfo.o $(SRCS)
+		$(CXX) $(TESTOBJS) $(UTILS)/cpuinfo.o $(MARCHFLAG) $(CXXFLAGS) -lgtest_main $(GTESTLDFLAGS) -o testexe
 
 bench: $(BENCHDIR)/main.cpp $(SRCS) $(UTILS)/cpuinfo.o
 		$(CXX) $(BENCHDIR)/main.cpp $(CXXFLAGS) $(UTILS)/cpuinfo.o $(MARCHFLAG) -o benchexe
 
+meson:
+	meson setup --warnlevel 0 --buildtype release builddir
```

Comment:
The `release` buildtype is valid. But. Given you're already adding cflags to your targets, best to keep it `plain`. Otherwise meson will append non-obvious args/flags that could surprise folks.

To further insulate yourself from meson policies, you can look into adding the `--debug false` argument as well.

Fyi, should you ever need them, know that meson has finer-grained linker control arguments as well.

Reply:
Makes sense.